### PR TITLE
this is why we can't have nice things

### PR DESCRIPTION
**Pull request by SomeguyManperson**
:cl:  
bugfix: you can only have TWO borg expander modules per borg. You know what you did.
/:cl:

### CODE REVIEW

**Comment by ZombieNub** [comment]
I know what I did but in my defense the borg wanted it and how was I gonna say no?

**Comment by ToasterBiome** [comment]
delete this

**Comment by N3D6** [comment]
theos once again saves yog from fun

**Comment by ZombieNub** [comment]
In my opinion expansion stacking should remain. It's funny and I'll let others have fun with it if they so please.

**Comment by SomeguyManperson** [comment]
maybe stacking up to a limit but you can get so big right now you can obscure like the whole screen

**Comment by ToasterBiome** [comment]
> maybe stacking up to a limit but you can get so big right now you can obscure like the whole screen

I would say like 2-3 would be good for "meme" but also not game breakingly annoying

**Comment by TheLinuxOS** [comment]
I love how you say “theos saves yogs from fun” when people were literally exploiting it to be as big as the escape shuttle

**Review by alexkar598 — CHANGES_REQUESTED** [review]
1) Changelog is incorrect, its twice not once
2) Changelog is incorrect, it would be a tweak not a bugfix at this point since the intended behaviour is once not twice
3) `hasExpanded` needs to be changed from being a boolean to being a number at the following lines:
    - https://github.com/yogstation13/Yogstation/blob/master/code/game/objects/items/robot/robot_upgrades.dm#L708 (R.hasExpanded--)
    - https://github.com/yogstation13/Yogstation/blob/master/code/modules/mob/living/silicon/robot/robot.dm#L87 (`var/hasExpanded = 0`)
    - https://github.com/yogstation13/Yogstation/blob/master/code/modules/mob/living/silicon/robot/robot.dm#L1027 (`hasExpanded--`)
4) Ideally, `hasExpanded` should be renamed to something else so that its less confusing since its no longer a boolean. Perhaps `expansions`?

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -706,9 +706,9 @@
 /obj/item/borg/upgrade/expand/deactivate(mob/living/silicon/robot/R, user = usr)
 	. = ..()
 	if (.)
-		if (R.hasExpanded)
-			R.hasExpanded = FALSE
+		while(R.expansion_count)
 			R.resize = 0.5
+			R.expansion_count--
 			R.update_transform()
```
Comment (alexkar598) [inline code/game/objects/items/robot/robot_upgrades.dm:712]: I'm not 100% a fan of using a while loop here as its technically not needed but I suppose it prevents bugs from turning borgs invisible by rounding their size down to 0

**Review by alexkar598 — APPROVED** [review]

**Pull request merged**